zulipterminal/core.py

```diff
@@ -83,20 +83,33 @@ def update_screen(self) -> None:
     def draw_screen(self, *args: Any, **kwargs: Any) -> None:
         self.loop.draw_screen()
 
-    def show_help(self) -> None:
+    def show_pop_up(self, to_show: Any, display_text: str) -> None:
         cols, rows = self.loop.screen.get_cols_rows()
-        help_view = HelpView(self)
         self.loop.widget = urwid.Overlay(
-            urwid.LineBox(help_view,
-                          title="Help Menu (up/down scrolls)"),
+            urwid.LineBox(to_show,
+                          title=display_text),
             self.view,
             align='center',
             valign='middle',
             # +2 to both of the following, due to LineBox
-            width=help_view.width+2,
-            height=min(3*rows//4, help_view.number_of_actions)+2
+            width=to_show.width+2,
+            height=to_show.height+2 if hasattr(to_show, 'height')
+                  else min(3*rows//4, to_show.number_of_actions)+2
         )
 
+    def show_help(self) -> None:
+        help_view = HelpView(self)
+        self.show_pop_up(help_view, "Help Menu (up/down scrolls)")
+
+    def show_msg_info(self, msg: Any) -> None:
+        msg_info_view = MsgInfoView(self, msg)
+        self.show_pop_up(msg_info_view,
+                         "Message Information (up/down scrolls)")
```

Comment:
This refactor commit is good to see :+1: As you may have seen in Tim's comment in #learning, it can be clearer if you adapt the code first (like this), then add the feature in a follow-up.